app/js/S3PersistorManager.js

```diff
@@ -240,9 +249,49 @@ function _wrapError(error, message, params, ErrorType) {
   }
 }
 
-function _defaultOptions() {
-  const options = {
-    credentials: {
+const _clients = {}
+
+function _client(bucket) {
```

Comment:
pedantic comment you can ignore, it isn't very clear what `_client()` does, would `_getClient` be a touch more descriptive?

Reply:
I've always had a soft preference for omitting `get` where possible, e.g. `foo.client()`, `foo.setClient(bar)`

Although in this case, I think it might be more descriptive to call it `_clientForBucket(bucket)`

It's definitely not the thing I reserve my strongest opinions for, though.

Reply:
Well, since we're talking about it... I usually try to make all function names verbs, so something like `getClientForBucket`.

It may also be worth noting that it's now relatively easy to define custom getters/setters in JS, e.g. to make `foo.bar = foo.bat` work even if `bar` and `bat` have some logic in them, which can cut down on a lot of the `get` and `set` noise for really simple getters and setters, which is I think one of the motivations for avoiding `get`.